admin/app.js

```diff
@@ -32,7 +32,34 @@ const logger = require('./services/log.service').getLogger()
 const sqlService = require('./services/data-access/sql.service')
 const app = express()
 setupLogging(app)
-sqlService.initPool()
+
+/**
+ * Sleep in ms
+ * @param ms - milliseconds
+ * @return {Promise}
+ */
+function sleep (ms) {
+  return new Promise(resolve => {
+    setTimeout(resolve, ms)
+  })
+}
+
+// Initialise the Database Connection pool
+;(async function () {
+  try {
+    logger.debug('Attempting to initialise the connection pool')
+    await sqlService.initPool()
+  } catch (error) {
+    logger.alert('Failed to connect to the database', error)
+    // The initial probe connection was not able to connection: the DB is not available.  This will cause all
+    // connections in the connection pool to be initialised to closed connections. By pausing, we allow time for the
+    // db to become available.  When run in a docker container the PM2 process manager will restart the process, and
+    // hopefully the DB will be up by then.
+    await sqlService.drainPool()
+    await sleep(30000)
```

Comment:
why 30secs sleep?

Reply:
@jon-shipley can we make this configurable please?  i would rather we have the ability to act quickly and set this to a lower value if ever necessary

Reply:
30 seconds is a wait time to allow the database to come back online.  It also prevents resource over-utilisation should the database not be available.

I will make it configurable.